File: Actito.UserInbox.iOS.Binding.Bridge/Sources/Models/ActitoUserInboxItem.swift
import Foundation
import ActitoUserInboxKit
import ActitoBinding

@objc
public class ActitoUserInboxItem : NSObject {
    @objc public let inboxItemId: String
    @objc public let notification: ActitoNotification
    @objc public let time: Date
    @objc public let opened: Bool
    @objc public let expires: Date?

    @objc public init(inboxItemId: String, notification: ActitoNotification, time: Date, opened: Bool, expires: Date?) {
        self.inboxItemId = inboxItemId
        self.notification = notification
        self.time = time
        self.opened = opened
        self.expires = expires
    }

    public convenience init(from item: ActitoUserInboxKit.ActitoUserInboxItem) {
        self.init(
            inboxItemId: item.id,
            notification: ActitoBinding.ActitoNotification(from: item.notification),
            time: item.time,
            opened: item.opened,
            expires: item.expires
        )
    }

    public func toNative() -> ActitoUserInboxKit.ActitoUserInboxItem {
        return ActitoUserInboxKit.ActitoUserInboxItem(
            id: inboxItemId,
            notification: notification.toNative(),
            time: time,
            opened: opened,
            expires: expires
        )
    }
}

File: Actito.UserInbox.Core/Models/ActitoUserInboxItem.cs
using ActitoSdk.Core.Models;

namespace ActitoSdk.UserInbox.Core.Models;

/// <summary>
/// Represents an item in the Actito user inbox.
/// </summary>
/// <remarks>
/// An <see cref="ActitoUserInboxItem"/> contains a notification and metadata about its
/// read state within the inbox. Inbox items can optionally have an expiration date.
/// </remarks>
public class ActitoUserInboxItem
{
    /// <summary>
    /// Unique identifier of the inbox item.
    /// </summary>
    public string Id { get; }

    /// <summary>
    /// Notification associated with this inbox item.
    /// </summary>
    public ActitoNotification Notification { get; }

    /// <summary>
    /// Timestamp indicating when the item was received.
    /// </summary>
    public DateTime Time { get; }

    /// <summary>
    /// Indicates whether the item has been opened by the user.
    /// </summary>
    public bool Opened { get; }

    /// <summary>
    /// Optional expiration timestamp of the item.
    /// </summary>
    public DateTime? Expires { get; }

    /// <summary>
    /// Constructor for <see cref="ActitoUserInboxItem"/>.
    /// </summary>
    public ActitoUserInboxItem(string id, ActitoNotification notification, DateTime time, bool opened,
        DateTime? expires)
    {
        Id = id;
        Notification = notification;
        Time = time;
        Opened = opened;
        Expires = expires;
    }
}
